lib/matplotlib/tests/test_ticker.py

```diff
@@ -457,7 +458,8 @@ class TestIndexFormatter:
                                           (2, 'label2'),
                                           (2.5, '')])
     def test_formatting(self, x, label):
-        formatter = mticker.IndexFormatter(['label0', 'label1', 'label2'])
+        with cbook._suppress_matplotlib_deprecation_warning():
+            formatter = mticker.IndexFormatter(['label0', 'label1', 'label2'])
```

Comment:
If we are deprecating, is there any reason to keep the test?

Reply:
I think the usual policy is to get rid of the test only when the API is deprecated too.

Reply:
I guess that makes sense because we can't break things until we can break them...